syntax = "proto3";

package upload;

option go_package = "/gen/upload";

service Upload {
    rpc StartUpload(UploadRequest) returns (UploadResponse) {}
    rpc UploadChunk(Chunk) returns (ChunkResponse) {}
}

message UploadRequest {
    string file_name = 1;
    string checksum = 2;
    int32 size = 3;
    // eventually we want to use some auth
    // string token = 4;
}

message UploadResponse {
    string file_id = 1;
}


message Chunk {
    string file_id = 1;
    bytes data = 2;
    bool final = 3;
} 

message ChunkResponse {
    bool success = 1;
    string message = 2;
}